src/circllhist.c

```diff
@@ -1637,3 +1641,42 @@ hist_approx_inverse_quantile(const histogram_t *hist, const double *in, int in_s
   }
   return 0;
 }
+
+histogram_t *
+hist_create_approximation_from_adhoc(histogram_approx_mode_t mode,
+                                     const histogram_adhoc_bin_t *bins,
+                                     size_t nbins, double sum) {
+  histogram_t *h = hist_alloc_nbins(nbins);
+  double last_spread = 0;
+  for(size_t i=0; i<nbins; i++) {
+    double a = bins[i].lower;
+    double b = bins[i].upper;
+    if(abs(a) > abs(b)) {
+      a = bins[i].upper;
+      b = bins[i].lower;
+    }
+    // chose a midpoint with minimal error
+    double m = 0;
+    switch(mode) {
+      case HIST_APPROX_HIGH:
+        m = bins[i].upper - fabs(bins[i].upper) / 1000; break;
+      case HIST_APPROX_LOW:
+        m = bins[i].lower + fabs(bins[i].lower) / 1000; break;
+      case HIST_APPROX_MID:
+        m = (a / (a + b)) * fabs(b-a) + bins[i].lower;
+        break;
+    }
+    // however if we're a (-inf,B) or an (A,+inf) bucket we need to me more clever
```

Comment:
s/me/be/

Custom bins layout?

Reply:
Thanks.  Support for consuming the "oddly binned"